tv_apps/fling-player/test/unit/connector_test.js

```diff
@@ -0,0 +1,301 @@
+/* global expect, Connector, castingMsgTemplate, MockPresentation */
+'use strict';
+
+requireApp('fling-player/test/unit/casting_message_template.js');
+requireApp('fling-player/test/unit/mock_presentation.js');
+requireApp('fling-player/bower_components/evt/index.js');
+requireApp('fling-player/js/casting_message.js');
+requireApp('fling-player/js/master_debug.js');
+requireApp('fling-player/js/connector.js');
+
+suite('fling-player/Connector', function() {
+
+  var exp, msg, spy, connector, presentation;
+
+  setup(function (done) {
+    var timer;
+    presentation = new MockPresentation();
+    connector = new Connector(presentation);
+    presentation._controller.start();
+    connector.init();
+
+    this.timeout(200);
+    timer = setInterval(() => {
+      if (connector._isInit && connector._isInitConnection) {
+        clearInterval(timer);
+        done();
+      }
+    });
+  });
+
+  test('should send message', function () {
+    msg = castingMsgTemplate.get().ack;
+    spy = sinon.spy(presentation.receiver._connection, 'send');
+    connector.sendMsg(msg);
+    assert.isTrue(spy.withArgs(JSON.stringify(msg)).calledOnce);
+  });
+
+  suite('reply ack', function () {
+
+    setup(function () {
+      msg = castingMsgTemplate.get().play;
+      spy = sinon.spy(connector, 'sendMsg');
+    });
+
+    test('should reply ack', function () {
+      exp = castingMsgTemplate.get().ack;
+      exp.seq = msg.seq;
```

Comment:
This looks more like the initial testing data contains the `seq` number... But actually it's generated by us.